packages/tupaia-web/src/layout/MapLayout/MapLayout.tsx

```diff
@@ -53,21 +44,36 @@ const MapOverlaySelector = styled.div`
   border-radius: 5px;
   background: #ff7428;
   opacity: 0.6;
+  position: absolute;
+`;
+
+// Position this absolutely so it can be placed over the map
+const TilePickerWrapper = styled.div`
+  position: absolute;
+  right: 0;
+  bottom: 0;
+  height: 100%;
 `;
 
 /**
  * This is the layout for the lefthand side of the app, which contains the map controls and watermark, as well as the map
  */
 
 export const MapLayout = () => {
+  const [activeTileSet, setActiveTileSet] = useState(TILE_SETS[0]);
+
+  const onTileSetChange = (tileSetKey: string) => {
+    setActiveTileSet(TILE_SETS.find(({ key }) => key === tileSetKey) as typeof TILE_SETS[0]);
+  };
   return (
     <Wrapper>
-      <MapControlsContainer>
-        <MapOverlaySelector />
-        <MapLegendWrapper>{/** This is where the map legend would go */}</MapLegendWrapper>
-      </MapControlsContainer>
-      <Map />
-      {/** This is where the tilepicker would go */}
+      {/** order here matters - Map needs to be first so any controls can go over the top of it */}
```

Comment:
I'm not sure if this comment still makes sense?
Overall this approach is much better. Thanks for updating it.